src/Compilers/CSharp/Test/Emit/CodeGen/ForeachTest.cs

```diff
@@ -1653,5 +1653,43 @@ static void Main()
 
             CompileAndVerify(source, expectedOutput: "0");
         }
+
+        [Fact]
+        public void Var_ExtensionGetEnumerator()
+        {
+            var source = """
+                using System.Collections.Generic;
+                class MyCollection<T>
+                {
+                    public readonly List<T> Items;
+                    public MyCollection(params T[] items) { Items = new(items); }
+                }
+                static class Extensions
+                {
+                    public static IEnumerator<T> GetEnumerator<T>(this MyCollection<T> c) => c.Items.GetEnumerator();
+                }
+                class Program
+                {
+                    static void Main()
+                    {
+                        MyCollection<int> x = new(1, 2, 3);
+                        int total = 0;
+                        foreach (var y in x)
+                            total += y;
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.RegularDefault.WithFeature("run-nullable-analysis", "never"));
```

Comment:
>.WithFeature("run-nullable-analysis", "never") [](http://example.com/codeflow?start=89&length=46)

Why is this needed?

Reply:
This is needed to ensure we don't set `ForEachLocalSymbol._type` ahead of the call to `InferTypeOfVarVariable` which we're trying to test here. That method calls `ForEachLoopBinder.InferCollectionElementType` which should allow extension method `GetEnumerator`.